### ISSUE TITLE
Rate limiting spec ambiguous

### ISSUE THREAD

**Issue by Borkason** [matrix-org/matrix-doc#2421]
The specs state for every API endpoint if it is rate-limited or not. Example:
![grafik](https://user-images.githubusercontent.com/2971735/73019678-5d1abe80-3e24-11ea-87cb-5edca9e93f90.png)
https://matrix.org/docs/spec/client_server/latest#post-matrix-client-r0-register

But later in the specs it is stated that ...

> Homeservers SHOULD implement rate limiting to reduce the risk of being overloaded. If a request is refused due to rate limiting, it should return a standard error response of the form.
https://matrix.org/docs/spec/client_server/latest#rate-limiting

Does 12 override the rate-limiting specs of the specific endpoints or is the latter just a recommendation?

**Comment by turt2live** [comment]
Rate limiting is just a recommendation. Some endpoints are intentionally not rate limited, but we've also missed defining rate limiting on a few.

**Comment by richvdh** [comment]
honestly it seems odd to me for the spec to take a position on what should be rate-limited or not. That's pretty much an implementation decision.

The spec might want to give advice on endpoints that should have higher or lower rate limits, but putting it next to "Requires auth" like that makes it look like gospel.

**Comment by Borkason** [comment]
> honestly it seems odd to me for the spec to take a position on what should be rate-limited or not. That's pretty much an implementation decision.

Yes, that's what I was thinking until I arrived at No.12 in the specs. Also, wouldn't it be preferrable that rat-limiting is done at the web server layer rather than the server? The way rate-limiting is documented right now also implies that rate-limiting should be done in the Matrix server-implementation.

How about changing it to:

>  Rate-limit recommendation: Yes|No.

or

> Rate-limited: Recommended/Not recommended.

**Comment by joepie91** [comment]
Related: #1370?
